src/plugin.validation.dns.azure/AzureEnvironment.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+
+namespace PKISharp.WACS.Plugins.ValidationPlugins
+{
+    public class AzureEnvironment
```

Comment:
This whole class seems like an overly complicated way to specify an URI. Being able to provide an URI would also enable people to use their private Azure services (i.e. Azure Stack). If your concern is that people won't know what their options are or what endpoint to specify, it's still possible to use the `ChooseFromList` functions to have the user select from a list of pre-defined strings, and perhaps a `null` or other 'semi-magic' answer could trigger a call to `RequestString` to get the custom URI.

Reply:
Do you happen to mean `ChooseFromMenu`? I'm not seeing any `ChooseFromList` in the `IInputService`.

I wasn't familiar with the FCLP library but I can work it out from existing code and implement a custom URI method.